### PR TITLE
update(panel): Adding disableParentScroll

### PR DESCRIPTION
**Pull request by gmoothart**
This adds a config to the panel which disables scrolling of the panel's
parent (attachTo) element, if set.

Part of #7683

### CODE REVIEW

**Comment by gmoothart** [comment]
PTAL @DerekLouie @ErinCoughlan @KarenParker @ThomasBurleson @jelbourn

**Comment by KarenParker** [comment]
Please add a test for this (dialog has an example of what you can test).

**Comment by gmoothart** [comment]
Added tests and fixed a z-index issue, PTAL @DerekLouie @ErinCoughlan @KarenParker @ThomasBurleson @jelbourn

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -935,6 +939,11 @@ MdPanelRef.prototype.detach = function() {
     return self._$q.resolve(self);
   };
 
+  if (this._restoreScroll) {
+    this._restoreScroll();
+    delete this._restoreScroll;
```
Comment (jelbourn) [inline src/components/panel/panel.js]: Just set to `null`; it accomplishes the same thing and is generally always faster.

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**